pkg/common-controller/snapshot_controller.go

```diff
@@ -986,6 +1001,10 @@ func (ctrl *csiSnapshotCommonController) updateSnapshotStatus(snapshot *crdv1.Vo
 			newStatus.RestoreSize = resource.NewQuantity(*size, resource.BinarySI)
 			updated = true
 		}
+		if (newStatus.Error == nil && volumeSnapshotErr != nil) || (newStatus.Error != nil && volumeSnapshotErr != nil && newStatus.Error.Time != nil && volumeSnapshotErr.Time != nil && &newStatus.Error.Time != &volumeSnapshotErr.Time) {
```

Comment:
new status should be updated unconditionally? so that we can clear any error in the volume snapshot object?

Reply:
Yes

Reply:
I am not sure I understand this if statement here: "&newStatus.Error.Time != &volumeSnapshotErr.Time"
and what if newStats.Error != nil but volumeSnapshotErr is nil? in this case, should newStatus.Error be updated to nil as well?